xlators/protocol/client/src/client-rpc-fops.c

```diff
@@ -2401,6 +2401,13 @@ client3_3_lk_cbk(struct rpc_req *req, struct iovec *iov, int count,
         goto out;
     }
 
+    xdata = dict_new();
+    if (xdata == NULL) {
+        rsp.op_errno = ENOMEM;
+        rsp.op_ret = -1;
+        goto out;
+    }
+
```

Comment:
This xdata is getting over-written in client_post_lk() right?

Reply:
Yeah, removed this dict creation from here and after the client_post_lk() directly appending the strict-locks value to the xdata now.